mesh/mesh.cpp

```diff
@@ -2555,82 +2556,110 @@ void Mesh::ReorderElements(const Array<int> &ordering, bool reorder_vertices)
 }
 
 
+void Mesh::GetEdgeLengths(const DSTable &v_to_v, Array<double> &lengths) const
+{
+   auto GetLength = [this](int i, int j)
+   {
+      double length = 0.;
+      if (Nodes == NULL)
+      {
+         const double *vi = vertices[i]();
+         const double *vj = vertices[j]();
+         for (int k = 0; k < spaceDim; k++)
+         {
+            length += (vi[k]-vj[k])*(vi[k]-vj[k]);
+         }
+      }
+      else
+      {
+         Array<int> ivdofs, jvdofs;
+         Nodes->FESpace()->GetVertexVDofs(i, ivdofs);
+         Nodes->FESpace()->GetVertexVDofs(j, jvdofs);
+         for (int k = 0; k < ivdofs.Size(); k++)
+         {
+            length += ((*Nodes)(ivdofs[k])-(*Nodes)(jvdofs[k]))*
+                      ((*Nodes)(ivdofs[k])-(*Nodes)(jvdofs[k]));
+         }
+      }
+      return length;
```

Comment:
This is length squared, which is fine for comparisons of length, and has the advantage of avoiding `sqrt`. However, the names of the functions should show that it is squared.

Reply:
Resolved in [e14459b](https://github.com/mfem/mfem/pull/3847/commits/e14459bfd057b1be1ba2ffa5ee19b97ebb085990)